src/test/regress/expected/multi_with_nested.out

```diff
@@ -0,0 +1,458 @@
+   2 |       2 | Thu Nov 23 06:01:08.148777 2017 |       2 |       4 |       2 |        
+(20 rows)
+
+-- nested CTEs should not be able to reference outer CTEs even if the query doesn't call CTE
+WITH users_events AS (
+  WITH users_events_2 AS (
+    SELECT
+        *
+    FROM
+        users_events
+  )
+  SELECT
+      *
+  FROM
+    users_table
+)
+SELECT 
+  * 
+FROM 
+  users_events
+ORDER BY
+  1, 2
+LIMIT 20;
+ERROR:  relation "users_events" does not exist
+LINE 6:         users_events
+                ^
+DETAIL:  There is a WITH item named "users_events", but it cannot be referenced from this part of the query.
+HINT:  Use WITH RECURSIVE, or re-order the WITH items to remove forward references.
+WITH users_events AS (
+  WITH users_events_2 AS (
+    SELECT
+        *
+    FROM
+        users_events
+  )
+  SELECT
+      *
+  FROM
+    users_table
+)
+SELECT 
+  * 
+FROM 
+  users_table
+ORDER BY
+  1, 2
+LIMIT 20;
+ERROR:  relation "users_events" does not exist
+LINE 6:         users_events
+                ^
+DETAIL:  There is a WITH item named "users_events", but it cannot be referenced from this part of the query.
+HINT:  Use WITH RECURSIVE, or re-order the WITH items to remove forward references.
+-- Even if it is not called, nested CTEs should not be able to reference outer CTE
+WITH nested1 AS (
+  WITH nested2_1 AS (
+    SELECT * FROM users_table
+  ),
+  nested2_2 AS (
+    SELECT * FROM nested1
+  )
+  SELECT * FROM nested2_1
+)
+SELECT * FROM nested1;
+ERROR:  relation "nested1" does not exist
+LINE 6:     SELECT * FROM nested1
+                          ^
+DETAIL:  There is a WITH item named "nested1", but it cannot be referenced from this part of the query.
+HINT:  Use WITH RECURSIVE, or re-order the WITH items to remove forward references.
+-- Outer CTE cannot reference the inner CTE in another CTE.
+WITH nested1 AS (
+  WITH nested1_1 AS (
+    SELECT * FROM users_table
+  )
+  SELECT * FROM nested1_1
+),
+nested2 AS (
+  SELECT * FROM nested1_1
+)
+SELECT * FROM nested2;
+ERROR:  relation "nested1_1" does not exist
+LINE 8:   SELECT * FROM nested1_1
+                        ^
+-- inner CTE cannot reference another inner CTE if they are in two different outer CTEs
+WITH nested1 AS (
+  WITH nested1_1 AS (
+    SELECT * FROM users_table
+  )
+  SELECT * FROM nested1_1
+),
+nested2 AS (
+  WITH nested2_1 AS (
+    SELECT * FROM nested1_1
+  )
+  SELECT * FROM nested2_1
+)
+SELECT * FROM nested2;
+ERROR:  relation "nested1_1" does not exist
+LINE 9:     SELECT * FROM nested1_1
+                          ^
+-- inner CTE cannot reference the outer CTE
+WITH nested1 AS (
+  WITH nested1_1 AS (
+    SELECT * FROM nested2
+  )
+  SELECT * FROM nested1_1
+),
+nested2 AS (
+  WITH nested2_1 AS (
+    SELECT * FROM nested1_1
+  )
+  SELECT * FROM nested2_1
+)
+SELECT * FROM nested2;
+ERROR:  relation "nested2" does not exist
+LINE 3:     SELECT * FROM nested2
```

Comment:
note when it says `LINE ...` it's a postgres syntax error, which never hits any Citus code.

Reply:
Then I am removing all of the tests related to access from inner to outer CTEs which are all ended up with this error.